server.py

```diff
@@ -1,10 +1,230 @@
+from flask_sqlalchemy import SQLAlchemy
+import sqlalchemy
+from sqlalchemy.dialects.postgresql.base import UUID
+from uuid import uuid4
+import secrets
+from sendgrid import SendGridAPIClient
+from sendgrid.helpers.mail import Mail
+from flask_cors import CORS, cross_origin
 
 app = Flask(__name__)
+bcrypt = Bcrypt(app)
+cors = CORS(app)
+
+db = SQLAlchemy(app)
+app.config['SQLALCHEMY_DATABASE_URI'] = os.environ.get('DATABASE_URL').replace("://", "ql://", 1)
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+app.config['ACCESS_TOKEN_SECRET'] =  os.environ.get('ACCESS_TOKEN_SECRET')
+app.config['REFRESH_TOKEN_SECRET'] =  os.environ.get('REFRESH_TOKEN_SECRET')
+
+class Users(db.Model):
+    __tablename__ = 'users'
+    id = db.Column(UUID(as_uuid=True), primary_key=True, unique=True, server_default=sqlalchemy.text("uuid_generate_v4()"),)
+    email = db.Column(db.String(255), unique=True, nullable=False)
+    password = db.Column(db.String(255), nullable=False)
+    created_at = db.Column(db.DateTime(), nullable=False, server_default=db.func.now())
+    updated_at = db.Column(db.DateTime(), nullable=False, server_default=db.func.now(), server_onupdate=db.func.now())
+    reset_password_hash = db.Column(db.String(64), unique=True, nullable=False)
+    reset_password_last_requested_at = db.Column(db.DateTime(), nullable=True)
+
+def __init__(self, email, password, created_at, updated_at, reset_password_hash, reset_password_last_requested_at):
+    self.email = email
+    self.password = password    
+    self.created_at = created_at
+    self.updated_at = updated_at
+    self.reset_password_hash = reset_password_hash
+    self.reset_password_last_requested_at = reset_password_last_requested_at
 
 @app.route('/', methods=['POST', 'GET'])
 def hello_world():
     return jsonify({'success': 'Hello, World!'}), 200    
 
+def get_data_by_email(email):
+    count = db.session.query(Users).filter(Users.email == email).count()
+    if count <= 0:
+        return None
+    if count == 1:
+        user_details = db.session.query(Users).filter(Users.email == email)
+        for i in user_details:
+            return i.__dict__ 
+    if count > 1:
+        return {'error':'voilates the unique ability!'}, 400
+
+def token_valid_check(token):
+    count = db.session.query(Users).filter(Users.reset_password_hash == token).count()
+    if count <= 0:
+        return None
+    if count == 1:
+        user_details = db.session.query(Users).filter(Users.reset_password_hash == token)
+        for i in user_details:
+            return i.__dict__ 
+    if count > 1:
+        return {'error':'voilates the unique ability!'}, 400
+
+def send_reset_password_mail(email, token):
+    from_email = os.environ.get('SENDGRID_EMAIL')
+    to_email = email
+    subject = 'Reset password link'
+    content = 'To reset your password'
+    html_content = 'To reset your password <a href="https://backend-notion-clone.herokuapp.com/reset-password/%s">click here</a>' % token
```

Comment:
url should not be hard-coded; it can be different for each environments (staging, production)